hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/choose/algorithms/HealthyPipelineChoosePolicy.java

```diff
@@ -30,7 +31,11 @@
  */
 public class HealthyPipelineChoosePolicy implements PipelineChoosePolicy {
 
-  private PipelineChoosePolicy randomPolicy = new RandomPipelineChoosePolicy();
+  private PipelineChoosePolicy randomPolicy;
+
+  public HealthyPipelineChoosePolicy(NodeManager nodeManager) {
+    randomPolicy = new RandomPipelineChoosePolicy(nodeManager);
+  }
```

Comment:
In addition to compatibility, falling back to the no-arg constructor would also let us avoid adding this.